packages/omg-js-childchain/src/rpc/rpcApi.js

```diff
@@ -13,46 +13,69 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License. */
 
-const fetch = require('node-fetch')
+const request = require('request-promise')
 const debug = require('debug')('omg.childchain.rpc')
 const JSONBigNumber = require('json-bigint')
 
 class RpcError extends Error {
-  constructor ({ code, description }) {
-    super(description || code)
-    this.code = code
+  constructor (data) {
+    super(data.description || data.code || data)
   }
 }
 
-async function get (url) {
-  return fetch(url).then(parseResponse)
+async function get ({ url, proxyUrl }) {
+  let fetch = request
+  if (proxyUrl) {
+    fetch = request.defaults({ proxy: proxyUrl })
+  }
+
+  try {
+    const res = await fetch.get(url)
+    return parseResponse(res)
+  } catch (err) {
+    throw new Error(err)
+  }
 }
 
-async function post (url, body) {
+async function post ({ url, body, proxyUrl }) {
   body.jsonrpc = body.jsonrpc || '2.0'
   body.id = body.id || 0
 
-  return fetch(url, {
-    method: 'POST',
-    headers: { 'Content-Type': 'application/json' },
-    body: JSON.stringify(body)
-  }).then(parseResponse)
+  let fetch = request
+  if (proxyUrl) {
+    fetch = request.defaults({ proxy: proxyUrl })
+  }
+
+  try {
+    const options = {
+      method: 'POST',
+      uri: url,
+      headers: { 'Content-Type': 'application/json' },
+      body,
+      json: true
+    }
+    const res = await fetch.post(options)
+    return parseResponse(res)
+  } catch (err) {
+    throw new Error(err)
+  }
 }
 
-async function parseResponse (resp) {
-  const body = await resp.text()
+async function parseResponse (res) {
+  const body = JSON.stringify(res)
   let json
   try {
     // Need to use a JSON parser capable of handling uint256
     json = JSONBigNumber.parse(body)
   } catch (err) {
-    throw new Error('Unknown server error')
+    throw new Error(err.message)
```

Comment:
The reason for the 'Unknown server error' was because if there was an error in elixir-omg, it would return the error as a stack trace, which was not json. If that happened, then the JSON.parse failed with a format error, which wasn't very useful to the caller.
I'm not sure if 'Unknown server error' is any more useful, but at least the caller knows it's not their fault.

Also this may have changed on the elixir-omg side, maybe they return a 500 now